Server/mods/deathmatch/logic/CPedSync.cpp

```diff
@@ -191,73 +187,148 @@ void CPedSync::Packet_PedSync(CPedSyncPacket& Packet)
-                        pPed->SetInWater(pData->bIsInWater);
-
-                    // Send this sync
-                    pData->bSend = true;
+                    // Call the onPedDamage event
+                    CLuaArguments Arguments;
+                    Arguments.PushNumber(fDeltaHealth);
+                    pPed->CallEvent("onPedDamage", Arguments);
                 }
             }
         }
 
-        // Tell everyone
-        m_pPlayerManager->BroadcastOnlyJoined(Packet, pPlayer);
+        if (pData->ucFlags & 0x10)
+            pPed->SetArmor(pData->fArmor);
+
+        if (pData->ucFlags & 0x20)
+            pPed->SetOnFire(pData->bOnFire);
+
+        if (pData->ucFlags & 0x40)
+            pPed->SetInWater(pData->bIsInWater);
+
+        // Is it time to sync to everyone
+        bool bDoFarSync = llTickCountNow - pPed->GetLastFarSyncTick() >= g_TickRateSettings.iPedFarSync;
+
+        if (!bDoFarSync && pPed->IsNearPlayersListEmpty())
+            continue;
+
+        // Create a new packet, containing only the struct for this ped
+        CPedSyncPacket PedPacket(pData);
+        if (!&PedPacket)
+            continue;
+
+        if (bDoFarSync)
+        {
+            // Store the tick
+            pPed->SetLastFarSyncTick(llTickCountNow);
+            // Send to everyone
+            m_pPlayerManager->BroadcastOnlyJoined(PedPacket, pPlayer);
+            continue;
+        }
+
+        // Send to players nearby the ped
+        CSendList sendList;
+        for (auto iter = pPed->NearPlayersIterBegin(); iter != pPed->NearPlayersIterEnd(); iter++)
+        {
+            CPlayer* pRemotePlayer = *iter;
+            // If the syncer changes between UpdateNearPlayersList() he can be in the list, make sure we don't send to him
+            if (pRemotePlayer && pRemotePlayer != pPlayer)
+                sendList.push_back(pRemotePlayer);
+        }
+
+        if (!sendList.empty())
+            m_pPlayerManager->Broadcast(PedPacket, sendList);
+    }
+}
+
+void CPedSync::UpdateNearPlayersList()
+{
+    for (auto iter = m_pPedManager->IterBegin(); iter != m_pPedManager->IterEnd(); iter++)
+    {
+        CPed* pPed = *iter;
+        // Clear the list
+        pPed->ClearNearPlayersList();
+    }
+
+    for (auto iter = m_pPlayerManager->IterBegin(); iter != m_pPlayerManager->IterEnd(); iter++)
+    {
+        CPlayer* pPlayer = *iter;
+        if (!pPlayer->IsJoined() || pPlayer->IsBeingDeleted())
+            continue;
+
+        // Grab the camera position
+        CVector vecCameraPosition;
+        pPlayer->GetCamera()->GetPosition(vecCameraPosition);
+
+        // Do a query in the spatial database
+        CElementResult resultNearCamera;
+        GetSpatialDatabase()->SphereQuery(resultNearCamera, CSphere(vecCameraPosition, DISTANCE_FOR_NEAR_VIEWER));
+
+        for (auto iter = resultNearCamera.begin(); iter != resultNearCamera.end(); ++iter)
```

Comment:
`class CElementResult : public std::vector<CElement*>`
`for (CElement* pElement : resultNearCamera)` should work.

Reply:
It does, changed it.